File: UltimateERP/BusinessManager/Entities/entUsuario.vb
Public Class entUsuario

    Private m_empresa As String
    Private m_usuario As String
    Private m_descripcion As String
    Private m_clave As String
    Private m_email As String
    Private m_perfil As String
    Private m_aprobar_mn As Decimal
    Private m_aprobar_me As Decimal
    Private m_privilegios As Byte
    Private m_es_plantilla As Boolean
    Private m_emite_ticket As Boolean
    Private m_estado As String
    Private m_es_conforme As Boolean
    Private m_cambiar_clave As Boolean
    Private m_ejercicio As String = ""
    Private m_mes As Byte = 0
    Private m_periodo_actual As String = ""
    Private m_ejercicio_anterior As String = ""
    Private m_mes_anterior As Byte = 0

    Private m_serie_asignada As String
    Private m_almacen As String

    Public Sub New()
        Me.empresa = Nothing
        Me.usuario = Nothing
        Me.descripcion = Nothing
        Me.clave = Nothing
        Me.email = Nothing
        Me.perfil = Nothing
        Me.aprobar_mn = Nothing
        Me.aprobar_me = Nothing
        Me.privilegios = Nothing
        Me.es_plantilla = Nothing
        Me.emite_ticket = Nothing
        Me.estado = Nothing
        Me.es_conforme = Nothing
        Me.cambiar_clave = Nothing
        Me.ejercicio = Nothing
        Me.mes = Nothing
        Me.periodo_actual = Nothing
        Me.ejercicio_anterior = Nothing
        Me.mes_anterior = Nothing
        Me.serie_asignada = Nothing
        Me.almacen = Nothing
    End Sub

    Public Sub New(ByVal Usuario As entUsuario)
        With Usuario
            m_empresa = .empresa
            m_usuario = .usuario
            m_descripcion = .descripcion
            m_clave = .clave
            m_email = .email
            m_perfil = .perfil
            m_aprobar_mn = .aprobar_mn
            m_aprobar_me = .aprobar_me
            m_privilegios = .privilegios
            m_es_plantilla = .es_plantilla
            m_emite_ticket = .emite_ticket
            m_estado = .estado
            m_es_conforme = .es_conforme
            m_cambiar_clave = .cambiar_clave
            m_ejercicio = .ejercicio
            m_mes = .mes
            m_periodo_actual = .periodo_actual
            m_ejercicio_anterior = .ejercicio_anterior
            m_mes_anterior = .mes_anterior
            m_serie_asignada = .serie_asignada
            m_almacen = .almacen
        End With
    End Sub

    Public Property empresa() As String
        Get
            Return m_empresa
        End Get
        Set(ByVal value As String)
            m_empresa = value
        End Set
    End Property

    Public Property usuario() As String
        Get
            Return m_usuario
        End Get
        Set(ByVal value As String)
            m_usuario = value
        End Set
    End Property

    Public Property descripcion() As String
        Get
            Return m_descripcion
        End Get
        Set(ByVal value As String)
            m_descripcion = value
        End Set
    End Property

    Public Property clave() As String
        Get
            Return m_clave
        End Get
        Set(ByVal value As String)
            m_clave = value
        End Set
    End Property

    Public Property email() As String
        Get
            Return m_email
        End Get
        Set(ByVal value As String)
            m_email = IIf(String.IsNullOrEmpty(value), Space(1), value)
        End Set
    End Property

    Public Property perfil() As String
        Get
            Return m_perfil
        End Get
        Set(ByVal value As String)
            m_perfil = IIf(String.IsNullOrEmpty(value), usuario, value)
        End Set
    End Property

    Public Property aprobar_mn() As Decimal
        Get
            Return m_aprobar_mn
        End Get
        Set(ByVal value As Decimal)
            m_aprobar_mn = IIf(String.IsNullOrEmpty(value), 0, value)
        End Set
    End Property

    Public Property aprobar_me() As Decimal
        Get
            Return m_aprobar_me
        End Get
        Set(ByVal value As Decimal)
            m_aprobar_me = IIf(String.IsNullOrEmpty(value), 0, value)
        End Set
    End Property

    Public Property privilegios() As Byte
        Get
            Return m_privilegios
        End Get
        Set(ByVal value As Byte)
            m_privilegios = IIf(String.IsNullOrEmpty(value), 4, value)
        End Set
    End Property

    Public Property es_plantilla() As Boolean
        Get
            Return m_es_plantilla
        End Get
        Set(ByVal value As Boolean)
            m_es_plantilla = IIf(String.IsNullOrEmpty(value), False, value)
        End Set
    End Property

    Public Property emite_ticket() As Boolean
        Get
            Return m_emite_ticket
        End Get
        Set(ByVal value As Boolean)
            m_emite_ticket = IIf(String.IsNullOrEmpty(value), False, value)
        End Set
    End Property

    Public Property estado() As String
        Get
            Return m_estado
        End Get
        Set(ByVal value As String)
            m_estado = IIf(String.IsNullOrEmpty(value), "INA", value)
        End Set
    End Property

    Public Property es_conforme() As Boolean
        Get
            Return m_es_conforme
        End Get
        Set(ByVal value As Boolean)
            m_es_conforme = IIf(String.IsNullOrEmpty(value), False, value)
        End Set
    End Property

    Public Property cambiar_clave() As Boolean
        Get
            Return m_cambiar_clave
        End Get
        Set(ByVal value As Boolean)
            m_cambiar_clave = IIf(String.IsNullOrEmpty(value), False, value)
        End Set
    End Property

    Public Property ejercicio() As String
        Get
            Return m_ejercicio
        End Get
        Set(ByVal value As String)
            m_ejercicio = IIf(String.IsNullOrEmpty(value), Now.Year.ToString, value)
        End Set
    End Property

    Public Property mes() As Byte
        Get
            Return m_mes
        End Get
        Set(ByVal value As Byte)
            m_mes = IIf(String.IsNullOrEmpty(value), Now.Month, value)
        End Set
    End Property

    Public Property periodo_actual() As String
        Get
            Return m_periodo_actual
        End Get
        Set(ByVal value As String)
            m_periodo_actual = value
        End Set
    End Property

    Public Property ejercicio_anterior() As String
        Get
            Return m_ejercicio_anterior
        End Get
        Set(ByVal value As String)
            m_ejercicio_anterior = IIf(String.IsNullOrEmpty(value), Now.Year.ToString, value)
        End Set
    End Property

    Public Property mes_anterior() As Byte
        Get
            Return m_mes_anterior
        End Get
        Set(ByVal value As Byte)
            m_mes_anterior = IIf(String.IsNullOrEmpty(value), Now.Month, value)
        End Set
    End Property

    Public Property serie_asignada() As String
        Get
            Return m_serie_asignada
        End Get
        Set(ByVal value As String)
            m_serie_asignada = IIf(String.IsNullOrEmpty(value), "1", value)
        End Set
    End Property

    Public Property almacen() As String
        Get
            Return m_almacen
        End Get
        Set(ByVal value As String)
            m_almacen = IIf(String.IsNullOrEmpty(value), "001", value)
        End Set
    End Property

End Class

File: ModuloCobranzas/Lidoma_WebApplication/Entities/entUsuario.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using Microsoft.VisualBasic;

namespace Lidoma_WebApplication.Entities
{
    public class entUsuario
    {
        private string m_empresa;
        private string m_usuario;
        private string m_descripcion;
        private string m_clave;
        private string m_periodo_actual;
        private string m_email;
        private string m_perfil;
        private decimal m_aprobar_mn;
        private decimal m_aprobar_me;
        private int m_privilegios;
        private bool m_es_plantilla;
        private bool m_emite_ticket;
        private string m_estado;
        private bool m_es_conforme;
        private bool m_cambiar_clave;
        private string m_ejercicio;
        private int m_mes;
        private string m_ejercicio_anterior;
        private int m_mes_anterior;
        private string m_serie_asignada;
        private string m_almacen;

        public entUsuario()
        {
            this.empresa = null;
            this.usuario = null;
            this.descripcion = null;
            this.clave = null;
            this.periodo_actual = null;
            this.email = null;
            this.perfil = null;
            this.aprobar_mn = 0;
            this.aprobar_me = 0;
            this.privilegios = 0;
            this.es_plantilla = false;
            this.emite_ticket = false;
            this.estado = null;
            this.es_conforme = false;
            this.cambiar_clave = false;
            this.ejercicio = null;
            this.mes = 0;
            this.ejercicio_anterior = null;
            this.mes_anterior = 0;
            this.serie_asignada = null;
            this.almacen = null;
        }

        public string empresa
        {
            get { return m_empresa; }
            set
            {
                m_empresa = value;
            }
        }

        public string usuario
        {
            get { return m_usuario; }
            set
            {
                m_usuario = value;
            }
        }

        public string descripcion
        {
            get { return m_descripcion; }
            set
            {
                m_descripcion = value;
            }
        }

        public string clave
        {
            get { return m_clave; }
            set
            {
                m_clave = value;
            }
        }

        public string email
        {
            get { return m_email; }
            set
            {
                m_email = (String.IsNullOrEmpty(value) ? Strings.Space(1) : value);
            }
        }

        public string perfil
        {
            get { return m_perfil; }
            set
            {
                m_perfil = (String.IsNullOrEmpty(value) ? usuario : value);
            }
        }

        public decimal aprobar_mn
        {
            get { return m_aprobar_mn; }
            set
            {
                m_aprobar_mn = value == 0 ? 0 : value;
            }
        }

        public decimal aprobar_me
        {
            get { return m_aprobar_me; }
            set
            {
                m_aprobar_me = value == 0 ? 0 : value;
            }
        }

        public int privilegios
        {
            get { return m_privilegios; }
            set
            {
                m_privilegios = value == 0 ? 4 : value;
            }
        }

        public bool es_plantilla
        {
            get { return m_es_plantilla; }
            set
            {
                m_es_plantilla = value = false ? false : value;
            }
        }

        public bool emite_ticket
        {
            get { return m_emite_ticket; }
            set
            {
                m_emite_ticket = value = false ? false : value;
            }
        }

        public string estado
        {
            get { return m_estado; }
            set
            {
                m_estado = (String.IsNullOrEmpty(value) ? "INA" : value);
            }
        }

        public bool es_conforme
        {
            get { return m_es_conforme; }
            set
            {
                m_es_conforme = value = false ? false : value;
            }
        }

        public bool cambiar_clave
        {
            get { return m_cambiar_clave; }
            set
            {
                m_cambiar_clave = value = false ? false : value;
            }
        }

        public string ejercicio
        {
            get { return m_ejercicio; }
            set
            {
                m_ejercicio = (String.IsNullOrEmpty(value) ? DateTime.Now.Year.ToString() : value);
            }
        }

        public int mes
        {
            get { return m_mes; }
            set
            {
                m_mes = value == 0 ? DateTime.Now.Month : value;
            }
        }

        public string periodo_actual
        {
            get { return m_periodo_actual; }
            set
            {
                m_periodo_actual = (String.IsNullOrEmpty(value) ? String.Concat(DateTime.Now.Year.ToString(), DateTime.Now.Month.ToString("00")) : value);
            }
        }

        public string ejercicio_anterior
        {
            get { return m_ejercicio_anterior; }
            set
            {
                m_ejercicio_anterior = (String.IsNullOrEmpty(value) ? DateTime.Now.Year.ToString() : value);
            }
        }

        public int mes_anterior
        {
            get { return m_mes_anterior; }
            set
            {
                m_mes_anterior = value == 0 ? DateTime.Now.Month : value;
            }
        }

        public string serie_asignada
        {
            get { return m_serie_asignada; }
            set
            {
                m_serie_asignada = (String.IsNullOrEmpty(value) ? "1" : value);
            }
        }

        public string almacen
        {
            get { return m_almacen; }
            set
            {
                m_almacen = (String.IsNullOrEmpty(value) ? "001" : value);
            }
        }


    }
}